### DISCUSSION TITLE
faikin floods mqtt broker

### DISCUSSION THREAD

**Discussion by ataelim** [revk/xESP32-Faikout#480] [Q&A]
while looking for what the faikin actually subscribes to, i enabled the (mosquitto) broker's "log_all" and found that the faikin module
not only subscribes with \<prefix\>/\<MAC\>/# , \<prefix\>/\<appname\>/# AND \<prefix\>/\<hostname\>/#
but also publishes status messages with topic = \<MAC\> and payload = {"online":true,"target":22.00,"temp":22.42,"inlet":21.67,"liquid":19.41,"fanfreq":0.0,"mode":"off","fan":"high"} every 3 seconds , although "reporting" (status reporting period) is set to 60.
Furthermore, i cannot get the faikin to react to any of the set or command topics i am publishing to the broker. 
Maybe i completely misunderstand the documentation or something is simply wrong:
i am publishing with topic = \<prefix\>/\<hostname\> and payload = null (empty) , where \<prefix\> already contains the 'setting' string as set in the webui settings (and the faikin correctly subscribes to). it also does not respond when publishing with /* at the end of the topic, with "*" as the payload and with \<MAC\> or \<appname\> instead of \<hostname\> .
and it does not 'reset' or 'upgrade' when publishing the respective 'command' , neither when putting the command into the payload nor as suffix to the 'prefixcommand' setting in the web ui.
... i am at the end of wisdome here ... any ideas what is wrong ?
thanks !!

**Comment by revk** [comment]
I am not seeing status posts ever 3 seconds here, there are some settings to control some of the reports. I can look in to it if you like.

The subscribes look right.

But the `command` and `setting` stuff definitely works! Not `setting` always replies using `setting/`*MAC*

**Comment by ataelim** [comment]
Thanks a lot for quick reply !

i have set the *prefix*xxx values to my own values , so i can more easily
distinguish from other
MQTT devices on the same broker, e.g. prefixcommand = "faik/cmd" ,
prefixsetting="faik/set"
when looking at the mosquitto logs (see below) , the subscriptions for them
look correct.
(the appname is "Faikin" and the hostname is set to "Faikin1". ; the
'prefixapp' setting is false)

to confirm - given the mosquitto logs shown below, the correct topic and
payload to trigger a mqtt response with all settings
(i.e. request the Faikin to publish one ore more messages with the
settings) would be
topic=faik/set/Faikin/* or topic=faik/set/Faikin1/* or
topic=faik/set/DC5475EFA6D4/* and payload=empty (zero-lenght string),
while sending a command would be
topic=faik/cmd/Faikin and payload=reset

i am attaching a mqtt-monitor output with subscriptions to "faik/#" - where
it should send status reports every 60 seconds
(which it does NOT do - the message is received at 07:52 while the
timestamp in the message is 07:27 - those are
retained messages which the Faikin sent when (re)-connecting.
and to DC5475EFA6D4/# - where it actually publishes every 3 seconds :-)

final question: how to upgrade to latest firmware ? (the ota setting does
not seem to trigger and upgrade,
neither after 7 days nor after restart or powercycle ...)

thanks a million, have a great day and let me know, if you need more input,
Martin

---- mosquitto broker log: ----------
1726518310: New client connected from 10.221.74.178 as
Faikin:Faikin1_EFA6D4 (c1, k30).
1726518310:     faik/state/Faikin1
1726518310: Sending CONNACK to Faikin:Faikin1_EFA6D4 (0, 0)

1726518310: Received SUBSCRIBE from Faikin:Faikin1_EFA6D4
1726518310: Faikin:Faikin1_EFA6D4 0 *faik/cmd/DC5475EFA6D4/#*
1726518310: Sending SUBACK to Faikin:Faikin1_EFA6D4

1726518310: Received SUBSCRIBE from Faikin:Faikin1_EFA6D4
1726518310:     *faik/cmd/Faikin/#* (QoS 0)
1726518310: Faikin:Faikin1_EFA6D4 0 faik/cmd/Faikin/#
1726518310: Sending SUBACK to Faikin:Faikin1_EFA6D4

1726518310: Received SUBSCRIBE from Faikin:Faikin1_EFA6D4
1726518310:     *faik/cmd/Faikin1/#* (QoS 0)
1726518310: Faikin:Faikin1_EFA6D4 0 faik/cmd/Faikin1/#
1726518310: Sending SUBACK to Faikin:Faikin1_EFA6D4

1726518310: Received SUBSCRIBE from Faikin:Faikin1_EFA6D4
1726518310: Faikin:Faikin1_EFA6D4 0 *faik/set/DC5475EFA6D4/#*
1726518310: Sending SUBACK to Faikin:Faikin1_EFA6D4

1726518310: Received SUBSCRIBE from Faikin:Faikin1_EFA6D4
1726518310:     *faik/set/Faikin/#* (QoS 0)
1726518310: Faikin:Faikin1_EFA6D4 0 faik/set/Faikin/#
1726518310: Sending SUBACK to Faikin:Faikin1_EFA6D4

1726518310: Received SUBSCRIBE from Faikin:Faikin1_EFA6D4
1726518310:     *faik/set/Faikin1/# *(QoS 0)
1726518310: Faikin:Faikin1_EFA6D4 0 faik/set/Faikin1/#
1726518310: Sending SUBACK to Faikin:Faikin1_EFA6D4

On Tue, Sep 17, 2024 at 9:11 AM RevK ***@***.***> wrote:

> I am not seeing status posts ever 3 seconds here, there are some settings
> to control some of the reports. I can look in to it if you like.
>
> The subscribes look right.
>
> But the command and setting stuff definitely works! Not setting always
> replies using setting/*MAC*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/revk/ESP32-Faikin/discussions/480#discussioncomment-10667128>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BLKUUV7NMK4JOUGHDAKOKNLZW7IYZAVCNFSM6AAAAABOKDFTBGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANRWG4YTEOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

***@***.***:/usr/haus# python mqtt-monitor.py -b 10.221.75.66 -t "faik/#" -v yes
broker is  10.221.75.66
port is  1883
topic is  ['faik/#']
waiting to connect
connecting to broker 10.221.75.66 retry=  0
connected
will wait for all subs
starting 
Use CTRL+C to stop
Tue Sep 17 07:52:15 2024 faik/state/Faikin1={"ts":"2024-09-17T07:27:11Z","id":"DC5475EFA6D4","up":39736,"mqtt-up":36011}
Tue Sep 17 07:52:15 2024 faik/state/Faikin1/status={"protocol":"X50A","online":true,"home":22.3,"heat":false,"slave":false,"antifreeze":false,"flap":false,"fanrpm":0,"inlet":21.6,"liquid":19.4,"power":false,"mode":"C","temp":22.0,"fan":"5","autor":0.0,"autot":0.0,"auto0":"00:00","auto1":"00:00","autop":false}
Tue Sep 17 07:52:15 2024 faik/state/Faikin/Faikin1={"ts":"2024-09-16T16:02:44Z","id":"DC5475EFA6D4","up":false,"mqtt-up":58,"reason":"Settings changed"}
Tue Sep 17 07:52:15 2024 faik/state/Faikin/Faikin1/status={"protocol":"X50A","online":true,"home":22.5,"heat":false,"slave":false,"antifreeze":false,"flap":false,"fanrpm":0,"inlet":21.8,"liquid":19.7,"power":false,"mode":"C","temp":22.0,"fan":"5","autor":0.0,"autot":0.0,"auto0":"00:00","auto1":"00:00","autop":false}
^Cinterrrupted by keyboard
total number of messages analysed= 4  displayed  4
***@***.***:/usr/haus# 
***@***.***:/usr/haus# 
***@***.***:/usr/haus# python mqtt-monitor.py -b 10.221.75.66 -t "DC5475EFA6D4/#" -v yes
broker is  10.221.75.66
port is  1883
topic is  ['DC5475EFA6D4/#']
waiting to connect
connecting to broker 10.221.75.66 retry=  0
connected
will wait for all subs
starting 
Use CTRL+C to stop
Tue Sep 17 07:53:08 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:16 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:22 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:25 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:28 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:31 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:37 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:40 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:46 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:52 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:55 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:53:58 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:01 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:01 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:04 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:07 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:10 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:13 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:16 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:19 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:22 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:31 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:34 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:40 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:46 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:49 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:52 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:54:58 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:01 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:01 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:04 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:07 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:13 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:16 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:22 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:25 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:28 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:31 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:34 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:37 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:40 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:46 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:49 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:55 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:55:58 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:01 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:01 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:04 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:07 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:13 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:16 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:19 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:25 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:28 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:31 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.58,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:34 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.33,"fanfreq":0.0,"mode":"off","fan":"high"}
Tue Sep 17 07:56:37 2024 DC5475EFA6D4={"online":true,"target":22.00,"temp":22.24,"inlet":21.48,"liquid":19.23,"fanfreq":0.0,"mode":"off","fan":"high"}
^Cinterrrupted by keyboard
total number of messages analysed= 58  displayed  58
***@***.***:/usr/haus#

**Reply by revk** [reply]
OK, they should work, but the design was originally not for `/` in those prefixes, so that could be the issue. I believe they should work though.

Can test with the default prefixes and see how that behaves. Is it still then doing the MAC messages all the time? Do they work?

Also, the new code uses `topic` not `prefix`, so make sure you are on latest beta release anyway. I need to issue a release, but may be last release code does not have the latest logic for prefixes with `/`.

**Reply by ataelim** [reply]
allright - i experimented with a separate mqtt broker (to keep other household members happy :-)) and found:
+ the repetitive reporting on the \<MAC\> topic is caused by enabling **Homeassistant** flag - once i cleared that, the 'flooding' stopped.
+ when changing the prefixes to **not** contain "/" - the 'set' with "*" at the end of topic works , triggering a publish of 2 messages: \<state\>/\<hostname\> containing some information and \<state\>/\<hostname\>/status with another bunch of information.

i am struggling with updating the ESP32 over the air - the 'command' topics still dont have an effect (neither 'reset' nor 'upgrade').
i will have to climb the ceiling, remove the ESP module , connect to some development machine using serial line and then flash the new firmware.

**Reply by revk** [reply]
The web page allows you to update and select beta (unless this is your own build).

I'd get on to latest code.

**Reply by ataelim** [reply]
some observations:
the "status report period" seems to be minutes (and not seconds ...) this is why i had the impression that there were no status updates at all.
whenever the 'settings' ui is entered, a 'status' mqtt message is published with topic=\<status-prefix-topic\>/\<hostname\> and payload={ts,id,up,mqtt-up,mem,spi}

**Reply by revk** [reply]
`reporting` is seconds, default `60`. So should be once a minute.

**Reply by ataelim** [reply]
currently ( Version Faikin-S3-MINI-N4-R2: 4945ab0 2024-02-28T15:54:01) , mqttx monitor shows unsolicited messages from Faikin every hour. I will need to flash latest firmware onto the esp32 before testing further.

**Reply by wurmtal868** [reply]
The "inlet" temperature in the log above is constantly changing.
Maybe every change is causing a new message.
